File: Cpp/565_Array_Nesting.cpp
class Solution {
public:
    int arrayNesting(vector<int>& nums) {
        int result = 0;
        vector<bool> visit(nums.size(), false);
        
        for(int i = 0; i<nums.size(); ++i){
            int count = 0;
            int j = i;
            while(!visit[j]){
                visit[j] = true;
                ++count;
                j = nums[j];
            }
            result = max(result, count);
        }
        
        return result;
    }
};

File: Kotlin/565_Array_Nesting.kt
class Solution {
    fun arrayNesting(nums: IntArray): Int {
        var result = 0
        val visit = BooleanArray(nums.size)
        
        for(i in nums.indices){
            var count = 0
            var j = i
            while(!visit[j]){
                visit[j] = true
                ++count
                j = nums[j]
            }
            result = maxOf(result, count)
        }
        
        return result
    }
}